wgengine/magicsock/magicsock.go

```diff
@@ -1727,7 +1727,16 @@ func (c *Conn) runDerpReader(ctx context.Context, derpFakeAddr netip.AddrPort, d
 		case derp.HealthMessage:
 			health.SetDERPRegionHealth(regionID, m.Problem)
 		case derp.PeerGoneMessage:
-			c.removeDerpPeerRoute(key.NodePublic(m), regionID, dc)
+			switch m.Reason {
+			case derp.PeerGoneReasonDisconnected:
+			case derp.PeerGoneReasonNotHere:
+				c.logf("magicsock: derp-%d does not know about peer %s, removing route",
```

Comment:
A convention in the Tailscale codebase: start the log message string with `[unexpected]` to signal "this shouldn't be happening. i.e. `c.logf("[unexpected] magicsock: derp-%s ...")`

That way we can look for unexpected events in debug logs and keep track of how much unexpected stuff is going on.